src/geometries/ParametricGeometry.js

```diff
@@ -74,28 +78,36 @@ function ParametricBufferGeometry( func, slices, stacks ) {
 
 			var u = j / slices;
 
-			var p = func( u, v );
-			vertices.push( p.x, p.y, p.z );
+			// vertex
+
+			p0 = func( u, v, p0 );
```

Comment:
But what if the user provides a `func` with just `u` and `v` parameter? I though that `p0 = func( u, v, p0 );` ensures backward compatibility.